### ISSUE TITLE
`no-useless-path-segments` does not work with require statements

### ISSUE THREAD

**Issue by 1pete** [benmosher/eslint-plugin-import#1127]
from rule's [`doc`](https://github.com/benmosher/eslint-plugin-import/blob/37bd5423276abbe6ee2245a3adc5d178fe16121b/docs/rules/no-useless-path-segments.md), it is described that it works with both import and require statements, but it actually work only with import statements

Given the following folder structure:
```
my-project
├── app.js
└── pages
    ├── about.js
    ├── contact.js
    └── index.js
```
tested code
```js
/**
 *  in my-project/app.js
 */
require('./../pages/about.js');
```
**Expectation**: example pattern should be considered problem.
**Actual**: it's not.

You may also resolve this issue by amend the doc, but I hope you don't do it this way. 😛
